plugins/modules/foreman_ansible_roles_facts.py

```diff
@@ -0,0 +1,180 @@
+- debug:
+    var: result
+
+- name: Get all imported Ansible Roles
+  foreman_ansible_roles:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "
+    action: "list"
+  register: result
+- debug:
+    var: result
+
+- name: Find the imported Ansible Role named example_role
+  foreman_ansible_roles:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "
+    action: "list"
+    search: "name=example_role"
+  register: result
+- debug:
+    var: result
+'''
+
+RETURN = '''
+ansible_roles:
+  description: Roles that are able to be imported on the proxy
+  returned: state = fetched
+  type: list
+'''
+
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule, _flatten_entity
+
+
+class ForemanAnsibleRolesFactsModule(ForemanAnsibleModule):
+    pass
+
+
+def main():
+
+    module = ForemanAnsibleRolesFactsModule(
+        foreman_spec = dict(
+            action=dict(default='fetch', choices=['fetch', 'list', 'show']),
+            proxy=dict(type='entity', flat_name='proxy_id', aliases=['smart_proxy'], resource_type='smart_proxies'),
+            search=dict(default=""),
+            full_details=dict(type='bool', aliases=['info'], default=False),
+            organization=dict(type='entity'),
+            location=dict(type='entity'),
+        ),
+
+        required_if = [
+            ['action', 'fetch', ['proxy']]
+        ],
+    )
+
+    resource = 'ansible_roles'
+
+    if module.foreman_params['action'] == 'show':
+        module.foreman_params['action'] = 'list'
+        module.foreman_params['full_details'] = True
+
+    with module.api_connection():
+        module.auto_lookup_entities()
+        params = _flatten_entity(module.foreman_params, module.foreman_spec)
+
+        if module.foreman_params['action'] == 'fetch':
+            resources = module.fetch_resource(resource, params)
+            resources = resources['results']['ansible_roles']
+
+        elif module.foreman_params['action'] == 'list':
+            # According to the APIDoc organization_id and location_id are valid
+            # parameters, but a "500 Internal Server Error" error is returned
+            # when an organization_id or location_id are present in the request
+            if 'organization_id' in params:
+                del params['organization_id']
+
+            if 'location_id' in params:
+                del params['location_id']
+
+            response = module.list_resource(resource, module.foreman_params.get('search'), params)
+
+            if module.foreman_params['full_details']:
+                resources = []
+                for found_resource in response:
+                    resources.append(module.show_resource(resource, found_resource['id'], params))
+            else:
+                resources = response
+
+        module.exit_json(ansible_roles=resources)
+
+if __name__ == '__main__':
+    main()
```

Comment:
```suggestion
    main()

```

Reply:
Oh, let's hope, this only introduces a single line feed at the end...

Reply:
Oh yeah. Newline at the end of the file is all that's needed here.

Reply:
Fixed.